interpreter/cling/lib/Utils/ASTImportSource.cpp

```diff
@@ -0,0 +1,189 @@
+#include "cling/Utils/ASTImportSource.h"
+
+using namespace clang;
+
+namespace cling {
+  namespace utils {
+    ASTImportSource::ASTImportSource(cling::Interpreter *parent_interpreter,
+                                     cling::Interpreter *child_interpreter) :
+      m_parent_Interp(parent_interpreter), m_child_Interp(child_interpreter) {
+    }
+
+    void ASTImportSource::ImportDecl(Decl *declToImport,
+                                     ASTImporter &importer,
+                                     DeclarationName &childDeclName,
+                                     DeclarationName &parentDeclName,
+                                     const DeclContext *childCurrentDeclContext) {
+
+      Decl *importedDecl = nullptr;
+      /* Don't do the import if we have a Function Template.
+       * Not supported by clang. */
+      if (!(declToImport->isFunctionOrFunctionTemplate() && declToImport->isTemplateDecl()))
+        importedDecl = importer.Import(declToImport);
+
+      if (importedDecl) {
+        /* Import this Decl in the map we own. Not used for now
+         * m_Decls_map[Name.getAsString()] = std::make_pair(importedDecl, declToImport); */
+        std::vector < NamedDecl * > declVector;
```

Comment:
You could use `std::vector < NamedDecl * > declVector{importedNamedDecl};` instead of the runtime `push_back()`.